aptos-move/parallel-executor/src/scheduler.rs

```diff
@@ -59,31 +66,43 @@ pub enum SchedulerTask<'a> {
 /// to 'ReadyToExecute(incarnation + 1)', allowing the scheduler to create an execution
 /// task for the next incarnation of the transaction.
 ///
-/// Status transition diagram when with no dependencies:
-/// ReadyToExecute(i) => --exactly once-- => Executing(i) => --transaction executed i-th time-- =>
-/// => Executed(i) => --validations happen-- =>
-/// 1. validation failures: --exactly once-- => Aborting(i) => => ReadyToExecute(i+1)
-/// 2. no validation failures: status remains Executed(i).
-///
 /// Status transition diagram:
 /// Ready(i)
 ///    |  try_incarnate (incarnate successfully)
-///    ↓                                                resume
-/// Executing(i) (pending for exactly one execution) ------------> Ready(i+1)
+///    |
+///    ↓         suspend (waiting on dependency)                resume
+/// Executing(i) -----------------------------> Suspended(i) ------------> Ready(i+1)
+///    |
 ///    |  finish_execution
 ///    ↓
 /// Executed(i) (pending for (re)validations)
+///    |
 ///    |  try_abort (abort successfully)
 ///    ↓                finish_abort
-/// Aborting(i) -------------------------------------------------> Ready(i+1)
+/// Aborting(i) ---------------------------------------------------------> Ready(i+1)
 ///
 enum TransactionStatus {
-    ReadyToExecute(Incarnation),
+    ReadyToExecute(Incarnation, Option<DependencyCondvar>),
     Executing(Incarnation),
+    Suspended(Incarnation, DependencyCondvar),
     Executed(Incarnation),
     Aborting(Incarnation),
 }
 
+impl PartialEq for TransactionStatus {
```

Comment:
where do we need equality check?

Reply:
This is for tests, should I just move it to each test file?

Reply:
maybe just add #[cfg(test)]

Reply:
oh, also used in debug asserts.